### PR TITLE
Update Scintilla to version 3.7.5

### LINKED ISSUE #1308: Regression: No syntax highlighting for bash lists with one item
Bash lists with only one string item are not applying syntax highlighting.
It worked correctly with an older version of Geany, but I cannot say when this regression was introduced.

Example:
```
#!/bin/bash

var1=('foo')
var2=('foo: works')
var3=('foo', 'bar')
```
Expected syntax highlighting:
https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/audacious
See line 13, 16-20, 22.

System information:
* Arch Linux
* Geany 1.29

### PR DESCRIPTION
**Pull request by b4n**
This now requires a C++11-capable compiler.

Closes #1308.

---
PR'd both for a Travis run and for you guys to give it a look as it changes the C++ dep.

### CODE REVIEW

**Comment by b4n** [comment]
Holy cow.  Travis only has GCC 4.6… I'm afraid we'll have to try to switch to the new Travis setup, which we didn't yet for various reasons I don't remember, but that might be obsolete by now.  Let's hope.  If anybody wanna try that, please :)

**Comment by elextr** [comment]
Well, I give up for now, none of my attempts to modify the travis.yml to specify the gcc version have validated [here](http://lint.travis-ci.org/).

Anybody else?

**Comment by codebrainz** [comment]
What if you just put `-std=c++0x` into the `CXXFLAGS` for the travis.yml file? GCC 4.6 has [fairly good C++11 support](https://gcc.gnu.org/gcc-4.6/cxx0x_status.html) and given that Scintilla uses pretty basic C++ features, it might be enough.

**Comment by elextr** [comment]
Well, specifying the distro as @codebrainz suggested worked for linux, but probably not surprisingly not for mingw.

**Comment by elextr** [comment]
Hmmm, the mingw problem seems to be GTK, the compiler is now the same as Linux version.

**Comment by elextr** [comment]
The problem appears to be that the install of gtk-dev packages is conditioned on `MINGW` being undefined, so GTK isn't found.  How did this ever work?

**Comment by b4n** [comment]
@elextr we download Windows GTK bundles (see the `cross-mingw-build.sh` script)

**Comment by elextr** [comment]
@b4n ok so we are back to why doesn't it find GTK? (at least thats what I remember as the problem, can't access the logs any more, maybe Travis is borked)

**Comment by b4n** [comment]
@elextr I don't know, and that's why I tried to add some debugging info on why the check failed, but as you say apparently we can't get Travis logs, so that's useless for now…

**Comment by elextr** [comment]
GTK binary version= 2.10??????

**Comment by b4n** [comment]
> GTK binary version= 2.10??????

Seems to be normal, I have the same on, my Debian box

**Comment by elextr** [comment]
Ok, 2.10 is what Geany calls the ABI.  package config path problem then

**Comment by elextr** [comment]
Isn't it expanding the .pc files into _deps not _deps/lib/pkgconfig?

**Comment by elextr** [comment]
Hmm, didn't seem to list `_deps/lib/pkgconfig`

**Comment by b4n** [comment]
Nope, looks like it's ignoring `$PKG_CONFIG_PATH` for some reason…

**Comment by b4n** [comment]
Hacky but it works

**Comment by elextr** [comment]
Suppose will have to look at the content of the PR now :)

**Comment by eht16** [comment]
FWIW, the code builds and runs fine with MSYS2 on a native Windows box.

The changes look fine to me except the debug fun (`cat config.log` in scripts/cross-build-mingw.sh) and I cannot really judge the m4 code as it is just magic to me :(.

**Comment by b4n** [comment]
> FWIW, the code builds and runs fine with MSYS2 on a native Windows box.

Great, thanks for the heads up :)

> The changes look fine to me except the debug fun (`cat config.log` in scripts/cross-build-mingw.sh)

Do you see a problem with it?  I only output it when configure fails, which helps getting more info when running on Travis as we can't look at the files afterward.  I could move this to the Travis script, though it calls the build script and can't know really why it failed if it did, but it's not too much of a problem to try and cat a non-existing file on failure anyway, so I could do that.

>  and I cannot really judge the m4 code as it is just magic to me :(.

I didn't write it, it comes from the common m4 macro repository, AutoConf Archive, or rather the Debian stable version of it.  I however read it and it looks good to me.

**Comment by elextr** [comment]
> I didn't write it, it comes from the common m4 macro repository, AutoConf Archive, or rather the Debian stable version of it. I however read it and it looks good to me.

What!! you mean you didn't read all of Scintilla to find which C++11 features it uses and test for them ;-P

Will test "real soon now".

File: m4/ax_cxx_compile_stdcxx_11.m4 [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# ============================================================================
```
Comment (codebrainz) [inline m4/ax_cxx_compile_stdcxx_11.m4]: Isn't the newer macro to use `AX_CXX_COMPILE_STDCXX` with a wrapper for the specific standard like ` AX_CXX_COMPILE_STDCXX_11`? That's what I've been using the last several years in own projects, I thought this standalone one was deprecated/obsolete or something.

**Comment by elextr** [comment]
@b4n, Works for me, both GTK2 and GTK3.

**Comment by eht16** [comment]
>> The changes look fine to me except the debug fun (`cat config.log` in scripts/cross-build-mingw.sh)
> 
> Do you see a problem with it?  I only output it when configure fails, which helps getting more info when running on 

I missed the point that it is only printed on errors. So it is perfectly
fine.

What about merging this PR already? I guess in master we will get a
broader testing and since the current release cycle is going to end
soon, it could help to find any issues (if any).

(Even if the nightly builds for Windows will fail then. Either we can
fix this or we just live with the fact.)

**Comment by elextr** [comment]
@eht16 seems a good idea to commit, the debug commits can be taken out later if all is well.

**Comment by b4n** [comment]
> @eht16 <https://github.com/eht16> seems a good idea to commit, the debug
> commits can be taken out later if all is well.

 Agreed, but I need to backport
https://sourceforge.net/p/scintilla/code/ci/82cb780a04d12256758fc545b35779dc971d2de6/
because it's a real issue we already know about.  I'll try and do that
"soon", but myu current situation gives a varying mileage to "soon".

And that shows me I/we need to start going forward ands help maintaining
a Scintilla 3.x branch with fixes, at least until the new 4.x is viable
for everyone, and us in particular.

**Comment by kugel-** [comment]
@b4n Are we going to squeeze this in for 1.31?

**Comment by elextr** [comment]
@kugel- I doubt it, see @b4n comment above that a fix needs backporting.

File: m4/ax_cxx_compile_stdcxx_11.m4 (hunk 1 above)
Comment (b4n) [inline m4/ax_cxx_compile_stdcxx_11.m4]: Indeed, the one [currently on the on-line AutoConf archive](https://www.gnu.org/software/autoconf-archive/ax_cxx_compile_stdcxx_11.html) doesn't match the one in the corresponding Debian 8 package.  I guess I should indeed update to the latest upstream, although it's functionally the same when it's used to check for C++11 and not anything newer.

**Comment by b4n** [comment]
Actually the bug I'm talking about is already in master, nothing introduced by this PR.

**Comment by elextr** [comment]
@b4n, still, unless this fixes/adds something significant its probably better not to merge it three days before release :wink:

**Comment by elextr** [comment]
> Agreed, but I need to backport
https://sourceforge.net/p/scintilla/code/ci/82cb780a04d12256758fc545b35779dc971d2de6/
because it's a real issue we already know about.  I'll try and do that
"soon", but myu current situation gives a varying mileage to "soon".

> Actually the bug I'm talking about is already in master, nothing introduced by this PR.

Then can the above be a separate PR so we can merge this to get it tested?

**Comment by b4n** [comment]
Yeah, I think so.

**Comment by kugel-** [comment]
I'm using scintilla-375 + the above bugfix without problems thus far.

**Comment by elextr** [comment]
I have been using this without the bugfix no problems, though I don't use wrap and @b4n is happy and @kugel- is happy so merged.

**Pull request merged**